### PR TITLE
Support running batch prediction by launching a Dataflow job on GCP

### PR DESCRIPTION
**Pull request by yixinshi**
Note that since Dataflow on GCP doesn't support GPU (yet), we just create a regular CPU job instead.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2026)

### CODE REVIEW

**Comment by jlewi** [comment]
Thanks
Would you mind adding a jsonnet test for the prototype?
Examples:
https://github.com/kubeflow/kubeflow/tree/master/kubeflow/core/tests

**Comment by yixinshi** [comment]
Hi Jeremy, I found only the components defined with .libsonnet in kubeflow/core or kubeflow/xxx have unittest tests. It is easy to create unittest if the component defined in separate .libsonnet and .jsonnet. 

tf-batch-predict.jsonnet currently resides in kubeflow/example. No components in kubeflow/example/prototype/*.jsonnet has unittest, except tensorboard.jsonnot, in which the main code is in kubeflow/tensorboard/tensorboard.libsonnet.

So to create unittest for batch-predict, can I create a sub folder under kubeflow, i.e. kubeflow/batch-predict? And put tf-batch-predict.libsonnet there and tf-batch-predict.jsonnet in kubeflow/example? Let me know.

File: kubeflow/tf-batch-predict/tests/tf-batch-predict_test.jsonnet [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+              },
+            },
+          ],
+        },
+      },
+    },
+  }
+) &&
+
+std.assertEqual(
```
Comment (lluunn) [inline kubeflow/tf-batch-predict/tests/tf-batch-predict_test.jsonnet]: do you want to just test the changing part? e.g. for gpu, check .resources.limit

Comment (yixinshi) [reply kubeflow/tf-batch-predict/tests/tf-batch-predict_test.jsonnet]: Done

File: kubeflow/tf-batch-predict/prototypes/tf-batch-predict.jsonnet [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// @apiVersion 0.1
+// @name io.ksonnet.pkg.tf-batch-predict
+// @description TensorFlow batch-predict
+// @shortDescription A TensorFlow batch-predict job
+// @param name string Name to give to each of the components
+// @optionalParam numGpus number 0 number of GPUs to use
```
[the remark this answers is not in the document]
Comment (yixinshi) [reply kubeflow/tf-batch-predict/prototypes/tf-batch-predict.jsonnet]: Done. Also number does work.

File: kubeflow/tf-batch-predict/prototypes/tf-batch-predict.jsonnet [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// @param modelPath string 0 Path to the model directory
```
[the remark this answers is not in the document]
Comment (yixinshi) [reply kubeflow/tf-batch-predict/prototypes/tf-batch-predict.jsonnet]: Good catch! Done.

**Pull request merged**